music21/romanText/writeRoman.py

```diff
@@ -0,0 +1,348 @@
+        if self.title == 'Title unknown':
+            self._prepTitle()
+
+        self.analyst = analyst  # or self.score.metadata.analyst  # if supported in future
+
+        self.proofreader = proofreader  # or self.score.metadata.proofreader  # "
+
+        self.note = note
+
+        self.fileName = f'{self.composer}_-_{self.title}'
+        self.fileName = self.fileName.replace('.', '')
+        self.fileName = self.fileName.replace(':', '')
+        self.fileName = self.fileName.replace(' ', '_')
+
+        self.combinedList = [f'Composer: {self.composer}',
+                             f'Title: {self.title}',
+                             f'Analyst: {self.analyst}',
+                             f'Proofreader: {self.proofreader}']
+        if self.note:  # optional
+            self.combinedList.append(f'Note: {self.note}')
+
+        self.currentKey = ''
+
+    def _prepTitle(self):
+        '''
+        In the absence of a user-defined title,
+        attempt to prepare one from the score metadata looking at each of
+        the title, movementNumber and movementName attributes.
+        Failing that, a placeholder 'Unknown' stands in.
+        '''
+
+        if not self.score.metadata:
+            return
+
+        workingTitle = []
+
+        if self.score.metadata.title:
+            workingTitle.append(self.score.metadata.title)
+        if self.score.metadata.movementNumber:
+            workingTitle.append(f'- No.{self.score.metadata.movementNumber}:')  # Spaces later
+        if self.score.metadata.movementName:
+            if self.score.metadata.movementName != self.score.metadata.title:
+                workingTitle.append(self.score.metadata.movementName)
+
+        if len(workingTitle) > 0:
+            self.title = ' '.join(workingTitle)
+
+# ------------------------------------------------------------------------------
+
+    def prepList(self):
+        '''
+        Prepares a sequential list of text lines, with timeSignatures and romanNumerals
+        adding this to the (already prepared) metadata preamble ready for printing.
+        '''
+
+        for m in self.measureNumbers:
+
+            thisMeasure = self.container.measure(m)
+
+            # TimeSignatures
+            tsThisMeasure = [t for t in thisMeasure.getElementsByClass('TimeSignature')]
+            if tsThisMeasure:
+                firstTS = tsThisMeasure[0]
+                self.combinedList.append(f'\nTime Signature: {firstTS.ratioString}')
+                if len(tsThisMeasure) > 1:
+                    msg = 'further time signature change(s) unprocessed: ' \
+                          f'{[x.ratioString for x in tsThisMeasure[1:]]}'
+                    self.combinedList.append(f'Note: {msg}')
+
+            # RomanNumerals
+            measureString = ''  # Clear for each measure
+            rnsThisMeasure = [r for r in thisMeasure.getElementsByClass('RomanNumeral')]
+            if rnsThisMeasure:  # probably redundant
```

Comment:
Definitely redundant

```python
for rn in rnsThisMeasure:
```
handles this as it is

Reply:
definitely redundant.